.buildkite/scripts/changesets.sh

```diff
@@ -1,77 +1,62 @@
 #!/usr/bin/env bash
 
-# This script contains helper functions related to what should be run depending on Git changes
+set -euo pipefail
 
 OSS_MODULE_PATTERN="^[a-z0-9]+beat\\/module\\/([^\\/]+)\\/.*"
 XPACK_MODULE_PATTERN="^x-pack\\/[a-z0-9]+beat\\/module\\/([^\\/]+)\\/.*"
 
-are_paths_changed() {
-  local patterns=("${@}")
-  local changelist=()
-  for pattern in "${patterns[@]}"; do
-    changed_files=($(git diff --name-only HEAD@{1} HEAD | grep -E "$pattern"))
-    if [ "${#changed_files[@]}" -gt 0 ]; then
-      changelist+=("${changed_files[@]}")
-    fi
-  done
+definePattern() {
+  pattern="${OSS_MODULE_PATTERN}"
 
-  if [ "${#changelist[@]}" -gt 0 ]; then
-    echo "Files changed:"
-    echo "${changelist[*]}"
-    return 0
-  else
-    echo "No files changed within specified changeset:"
-    echo "${patterns[*]}"
-    return 1
+  if [[ "$beatPath" == *"x-pack/"* ]]; then
+    pattern="${XPACK_MODULE_PATTERN}"
   fi
 }
 
-are_changed_only_paths() {
-  local patterns=("${@}")
-  local changed_files=($(git diff --name-only HEAD@{1} HEAD))
-  local matched_files=()
-  for pattern in "${patterns[@]}"; do
-    local matched=($(grep -E "${pattern}" <<< "${changed_files[@]}"))
-    if [ "${#matched[@]}" -gt 0 ]; then
-      matched_files+=("${matched[@]}")
+defineExclusions() {
+  exclude="^$beatPath\/module\/(.*(?<!\.asciidoc|\.png))$"
+}
+
+defineFromCommit() {
+  local previousCommit
+  local changeTarget=${BUILDKITE_PULL_REQUEST_BASE_BRANCH:-$BUILDKITE_BRANCH}
+
+  previousCommit=$(git rev-parse HEAD^)
+
+  from=${changeTarget:+"origin/$changeTarget"}
+  from=${from:-$previousCommit}
+  from=${from:-$BUILDKITE_COMMIT}
```

Comment:
I am having a hard time understanding the logic above and what the above three definitions of the same variable do.

The way I What does this do exactly?

The way I read it ([docs](https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html)) is:

1. If `changeTarget` is unset or null, nothing is done (i.e. $from remains unset) otherwise it's set to `origin/$changeTarget`
2. if `$from` is unset or null then from becomes `$previousCommit` otherwise it remains as is
3. if `$from` is unset or null then from becomes `$BUILDKITE_COMMIT` otherwise it remains as is

Can we simplify this with easy to understand logic (and use conditionals where needed)?

For example I don't think that 3. will ever need to execute since `previousCommit=$(git rev-parse HEAD^)` and thus will always have a value, right?

Reply:
Simplified